docs/sni_mtls_pop_token_design.md

```diff
@@ -0,0 +1,192 @@
+# Overview
+
+Bearer tokens are vulnerable to theft. Proof-of-Possession (PoP) tokens mitigate this by binding tokens to a specific client certificate. mTLS PoP tokens enhance this security by using mutual TLS (mTLS) to ensure the token is tied to the certificate used for authentication.
```

Comment:
When you say "the token is tied to the certificate used for authentication" Are you saying that the SNI cert to authenticate the app is also used to init the HTTP MTLS connection? Are you not able to use 2 different certs here?

Reply:
added a note to clarify